Assets/BossRoom/Scripts/Client/Game/Action/MeleeActionFX.cs

```diff
@@ -27,6 +34,19 @@ public override bool Start()
             }
 
             base.Start();
+
+            // we can optionally have special particles that should play on the target. If so, add them now.
+            // (don't wait until impact, because the particles need to start sooner!)
+            if (Data.TargetIds != null && Data.TargetIds.Length > 0 && NetworkSpawnManager.SpawnedObjects.TryGetValue(Data.TargetIds[0], out var targetNetworkObj))
```

Comment:
We may need a null check here on the returned targetNetworkObject right?

Reply:
It occurred to me that we use this gross if-statement a lot (now with four separate `&&` checks!), and we could just add a method to `ActionRequestData`, something like `TryGetFirstTarget()`. That way we could just write 

`if (Data.TryGetFirstTarget(out var targetNetObj)` 

I was tempted to add it here, but this is a PR for particles, so only minimal code-changes are appropriate... perhaps a cleanup for another time.

ok

Reply:
Thanks!